land_grab_2/stl_dataset/step_2_5/get_cessions.py

```diff
@@ -0,0 +1,278 @@
+import logging
+import os
+from pathlib import Path
+from typing import List
+import warnings
+
+import pandas as pd
+import geopandas as gpd
+import numpy as np
+
+from land_grab_2.stl_dataset.step_1.constants import WGS_84
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+def area_join(
+    source_df: gpd.GeoDataFrame, target_df: gpd.GeoDataFrame, variables: List[str]
+) -> gpd.GeoDataFrame:
+    """
+    Join variables from source_df based on the largest intersection with feat-
+    ures from target_df. In case of a tie, pick the first joined record.
+    Implementation adapted from tobler:
+    https://pysal.org/tobler/_modules/tobler/area_weighted/area_join.html#area_join
+
+    Arguments:
+    source_df -- the GeoDataFrame containing source values
+    target_df -- the GeoDataFrame containing target values
+    variables -- string or list-like column(s) in source_df dataframe to join to
+                 target_df
+
+    Returns:
+    geopandas.GeoDataFrame -- target_df GeoDataFrame with joined variables as
+                              additional columns
+
+    """
+    for v in variables:
+        if v in target_df.columns:
+            raise ValueError(f"Column '{v}' already present in target_df.")
+
+    target_df = target_df.copy()
+    target_ix, source_ix = source_df.sindex.query(
+        target_df.geometry, predicate="intersects"
+    )
+    areas = (
+        target_df.geometry.values[target_ix]
+        .intersection(source_df.geometry.values[source_ix])
+        .area
+    )
+
+    main = []
+    for i in range(len(target_df)):
+        mask = target_ix == i
+        if np.any(mask):
+            main.append(source_ix[mask][np.argmax(areas[mask])])
+        else:
+            main.append(np.nan)
+
+    main = np.array(main, dtype=float)
+    mask = ~np.isnan(main)
+
+    for v in variables:
+        arr = np.empty(len(main), dtype=object)
+        arr[mask] = source_df[v].values[main[mask].astype(int)]
+        try:
+            arr = arr.astype(source_df[v].dtype)
+        except TypeError:
+            warnings.warn(
+                f"Cannot preserve dtype of '{v}'. Falling back to `dtype=object`.",
+            )
+        target_df[v] = arr
+
+    return target_df
+
+
+def join_counties_to_parcels(
+    parcels_gdf: gpd.GeoDataFrame, counties_gdf: gpd.GeoDataFrame
+) -> gpd.GeoDataFrame:
+    """
+    Join county name to parcels based on the largest spatial intersection.
+
+    Arguments:
+    parcels_gdf -- GeoDataFrame containing parcel geometries
+    counties_gdf -- GeoDataFrame containing county geometries
+
+    Returns:
+    GeoDataFrame -- STL parcels with county names joined as a new column
+    """
+    # Reproject counties_gdf to match parcels_gdf.
+    counties_gdf = counties_gdf.to_crs(parcels_gdf.crs)
+
+    out_gdf = area_join(
```

Comment:
And here's where we use `area_join` to find the county with the largest overlap for each parcel.